evm/vm/flavors/tester/__init__.py

```diff
@@ -0,0 +1,121 @@
+from cytoolz import (
+    assoc,
+)
+
+from eth_utils import (
+    reversed_return,
+)
+
+from evm import (
+    Chain,
+)
+
+from evm.vm.flavors import (
+    FrontierVM as BaseFrontierVM,
+    HomesteadVM as BaseHomesteadVM,
+    EIP150VM as BaseEIP150VM,
+)
+
+from evm.utils.chain import (
+    generate_vms_by_range,
+)
+
+
+def chfp_method_factory(parent_class):
+    def create_tester_header_from_parent(vm_class, parent_header, **header_params):
+        """
+        Creates and initializes a new block header from the provided
+        `parent_header`.
+        """
+        return parent_class.create_header_from_parent(
```

Comment:
I don't understand why we want to ignore the class on which the method was called (vm_class) and instead always use parent_class? Maybe it deserves a comment?